components/n-ui/ads/test/oAdsConfig.spec.js

```diff
@@ -103,8 +105,111 @@ describe('Config', () => {
 			const config = oAdsConfig(flags, 'article');
 			expect(config.gpt.zone).to.equal('testDfpSite/testDfpZone');
 		});
+
 	})
 
+	describe('lazyLoad viewportMargin', () => {
+
+		it('Should pass 0% when window.innerWidth is wider than 760px and adOptimizeLazyLoadSmall flag is defined', () => {
+			global.innerWidth = 760;
+			const flags = { get: (flagName) => {
+				switch (flagName) {
+					case 'adOptimizeLazyLoadSmall':
+					return '50';
+					break;
+					default:
+					return true;
+				}
+			}};
+			const config = oAdsConfig(flags, 'article');
+			expect(config.lazyLoad.viewportMargin).to.equal('0%');
+		});
+
+		it('Should pass 0% when window.innerWidth is less than 760px and adOptimizeLazyLoadSmall flag is undefined', () => {
+			global.innerWidth = 759;
+			const flags = { get: (flagName) => {
+				switch (flagName) {
+					case 'adOptimizeLazyLoadSmall':
+					return undefined;
+					break;
+					default:
+					return true;
+				}
+			}};
+			const config = oAdsConfig(flags, 'article');
+			expect(config.lazyLoad.viewportMargin).to.equal('0%');
+		});
+
+		context('when window.innerWidth is less than 760px and adOptimizeLazyLoadSmall flag is defined', () => {
+
+			beforeEach(() => {
+				global.innerWidth = 759;
+			});
+
+			afterEach(() => {
+				delete global.innerWidth;
+			});
+
+			it('Should pass 50% when the flag\'s value is 50', () => {
```

Comment:
as a time saving approach to future tests you can do something like the following
```
['50', '100', '150']
  .forEach(margin => {
    it(`Should pass ${margin}% when the flag\'s value is ${margin}`, () => {
        // write the test, using the margin value as appropriate
    })
  })
```
No need to do it for this PR (unless you want to), but thought I'd point it out as it's easy to think of tests as being somehow restricted in the way they're written, but actually you can use any js techniques you like to avoid code repetition.